scripts/pi-hole/php/add.php

```diff
@@ -18,47 +18,90 @@
 
 // Don't check if the added item is a valid domain for regex expressions. Regex
 // filters are validated by FTL on import and skipped if invalid
-if($type !== "regex") {
+$check_lists = ["white","black","audit"];
+if(in_array($list, $check_lists)) {
     check_domain();
 }
 
 // Escape shell metacharacters
-$domains = escapeshellcmd($_POST['domain']);
+$domains = explode(",",$_POST['domain']);
```

Comment:
The domains are sent as whitespace separated values (exactly as the user entered them).